dspace/modules/journal-submit/src/main/java/org/datadryad/submission/DryadGmailService.java

```diff
@@ -0,0 +1,213 @@
+                    .setAccessType("offline")
+                    .setApprovalPrompt("force")
+                    .addRefreshListener(new DataStoreCredentialRefreshListener(myUserID,myDataStoreFactory))
+                    .build();
+            myCredential = getMyCredential();
+        } catch (IOException e) {
+            throw new RuntimeException(e.toString());
+        }
+    }
+
+    // Default constructor builds a DryadGmailService for the dryad-journal-submit@gmail.com account
+    public DryadGmailService() {
+        this(new File(ConfigurationManager.getProperty("submit.journal.credential.dir")), ConfigurationManager.getProperty("submit.journal.clientsecrets").toString(), SCOPE, ConfigurationManager.getProperty("submit.journal.email"));
+    }
+
+    public void authorize (String code) {
+        GoogleAuthorizationCodeTokenRequest codeTokenRequest = myAuthCodeFlow.newTokenRequest(code);
+
+        try {
+            GoogleTokenResponse response = codeTokenRequest.setRedirectUri(GoogleOAuthConstants.OOB_REDIRECT_URI).execute();
+            myCredential = myAuthCodeFlow.createAndStoreCredential(response, myUserID);
+        } catch (IOException e) {
+            throw new RuntimeException("couldn't authorize: code was " + code + "\n" + e.toString());
+        }
+        return;
+    }
+
+    public String getAuthorizationURLString() {
+        return myAuthCodeFlow.newAuthorizationUrl().setRedirectUri(GoogleOAuthConstants.OOB_REDIRECT_URI).build();
+    }
+
+    public Credential getMyCredential() {
+        try {
+            return myAuthCodeFlow.loadCredential(myUserID);
+        } catch (IOException e) {
+            throw new RuntimeException("couldn't load credential: " + e.getMessage());
+        }
+    }
+
+    public Gmail getMyGmailService() {
+        if (myGmailService == null) {
+            // Create a new authorized Gmail API client
+            myGmailService = new Gmail.Builder(myHttpTransport, myJsonFactory, getMyCredential()).build();
+        }
+        return myGmailService;
+    }
+
+    public String getMyUserID() {
+        return myUserID;
+    }
+
+    public static String testMethod() throws IOException {
+        DryadGmailService dryadGmailService = new DryadGmailService();
+        // TEST CASE:
+        ArrayList<String> labels = new ArrayList<String>();
+        labels.add("test_gmail_api");
+
+        String result = "messages ";
+
+        List<Message> messages = dryadGmailService.listMessagesWithLabels(labels);
+        // Print ID of each Thread.
+        if (messages != null) {
+            for (Message m : messages) {
+                Message message = dryadGmailService.myGmailService.users().messages().get(dryadGmailService.myUserID, m.getId()).execute();
+                result = result + "\n" + message.getSnippet();
+            }
+        }
+        return result;
+    }
+
+    public List<Message> retrieveMessagesWithLabels (List<String> labels) throws IOException {
```

Comment:
If the labels don't actually exist, this will return all messages. I discovered this by trying to GET `/journal-submit/test` without having a `test_gmail_api` label defined.